website/public/js/services/socialServices.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+(function(){
+  angular
+    .module('habitrpg')
+    .factory('Social', socialFactory);
+
+  socialFactory.$inject = [];
+
+  function socialFactory() {
+
+    function loadWidgets() {
+      // Facebook
+      if (typeof FB === 'undefined') {
+        (function(d, s, id) {
+          var js, fjs = d.getElementsByTagName(s)[0];
+          if (d.getElementById(id)) return;
+          js = d.createElement(s); js.id = id;
+          js.src = "//connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.5";
+          fjs.parentNode.insertBefore(js, fjs);
+        }(document, 'script', 'facebook-jssdk'));
+      } else {
+        FB.XFBML.parse(); // http://stackoverflow.com/questions/29133563/
+      }
+
+      // Tumblr
+      $.getScript('https://assets.tumblr.com/share-button.js');
```

Comment:
It would be lovely if Tumblr has a way, like Twitter and Facebook, to rebuild widgets with the script already loaded. But I couldn't find one. So this results in the Tumblr script potentially getting loaded into memory multiple times over the course of the session. Maybe we can reverse-engineer `share-button.js` a bit? :P